### ISSUE TITLE
Hey man. Forked the project.

### ISSUE THREAD

**Issue by Skrimpton** [rogersavage/tkshortcut#1]
I just wanted to let you know I forked this for fun. 

I just quickly and poorly frankensteined together an object-oriented version from other stuff I had lying around.
Lots of bad python-code, and generally bad coding (I'm just a hobbyist script-kiddie), 
And I also maaaaay have had a glass or two of wine... 
But maybe there's some cool tkinter things or tricks in it you find useful.

Some new features are:
- Styling/black color scheme 
  - Uses a class. Just uncomment the line with ```self.styler``` in the ```__init__```  of MainWindow if you don't wanna use it.
- The entry fields are scrollable
  - ***Note:*** It doesn't like all diagonal scrolling on touchpad. 
  - It causes jittering because every other event is the opposite "button". 
  - And I've only tried touchpad scrolling on Linux. I think the or-check catches Windows events but not sure
- Shift up / down selects the word the text cursor is in or selects all text if something is selected
- You have undo/redo now
- It cobbles together the path from the executable's folder and the given filename
- You can only save the file if you have a filename
  - also it checks if the executable exists, but **not** if the directory can be written to 
- File is saved using ```with open```, and I added an error-message on failure 

Check it out it if you liked tkinter, and anyways ty for the fun :)

**Comment by rogersavage** [comment]
That's hilarious. This is great, thanks!

This is some completely unmitigated "dirty underwear" code that I decided to upload on a lark because I'm making a blog that I've barely shared with anyone yet, and here you found it randomly and put some actual effort into it. Amazing.

https://aaronsavage.net/2024/02/21/i-wish-id-learned-tkinter-sooner/

Maybe I'll kick this around a little more and we'll end up with something others will find useful. I didn't think anyone would even look at the code, let alone fork it. If I were going to take this seriously, it would need a few features that would take a bit more work, like editing existing shortcut files while preserving their unaffected fields, and some way to preview the icons. I'd also really like it if I could integrate it into a file manager. As a minimum, I could make it take a filename as an argument so people could "Open With..." to pre-populate the fields. It wouldn't be all that much work to make something legit, and you've got it off to a good start here.

Open source is a beautiful thing.

**Comment by Skrimpton** [comment]
Yeah, I was just farting around looking at tkinter stuff, cause I've been playing around with the kit lately myself.
Then your project popped up in most recent, and was just a perfect little idea for an evening and a real nice start.

Yeah, also the idea about letting it take a file as an argument is real good.
Reading the lines and finding the corresponding fields should be somewhere between easy and doable. 
But there are a lot of possible keys in a desktop-file I think (= a lot of fields?), so some way to dynamically add fields? 
Maybe using like a Combobox to select/add specific key-fields, and a Frame+Scrollbar for to put them in? 

Maybe a more hobby-friendly version is based around selecting some templates of "types of" desktop-files, then creating frames with fields for the templates?  
E.g. like one template for a game-shortcut, and one for a metadata-desktop-file used when making plasmoids, and so forth.  

Anyways something a little more fleshed out and dynamic like that could be a very nifty tool.

Making a ```~/.local/kservices5``` .desktop-file for user-added right-click-context-menu-entry that accepts a file-URL is at least real easy to make  on KDE. Dunno about other distros, but I suspect they're not putting up roadblocks 
(Well, maybe gnome does, cause it feels like an Apple-thing to do :P)

I've done some tweaking just now btw - discovered some blunders (there are for sure more), so I **highly** recommend a very critical read-through before you use it to build something serious. 

To illustrate the quality of my work and your possible foundation: It is literally made by a drunk fool, and there are poop jokes in the source code.
